packages/base/src/commands.ts

```diff
@@ -285,6 +290,177 @@ export function addCommands(
     ...icons.get(CommandIDs.newVectorTileEntry)
   });
 
+  commands.addCommand(CommandIDs.buffer, {
+    label: trans.__('Buffer'),
+    isEnabled: () => {
+      return tracker.currentWidget
+        ? tracker.currentWidget.model.sharedModel.editable
+        : false;
+    },
+    execute: async () => {
+      const layers = tracker.currentWidget?.model.sharedModel.layers ?? {};
+      const sources = tracker.currentWidget?.model.sharedModel.sources ?? {};
+
+      const layerOptions = Object.keys(layers).map(layerId => ({
+        value: layerId,
+        label: layers[layerId].name
+      }));
+
+      if (layerOptions.length === 0) {
+        console.warn('No layers available to buffer.');
+        return;
+      }
+
+      const schema = {
+        ...(formSchemaRegistry.getSchemas().get('Buffer') as IDict)
+      };
+
+      const model = tracker.currentWidget?.model;
+      const localState = model?.sharedModel.awareness.getLocalState();
+      if (!model || !localState || !localState['selected']?.value) {
+        return;
+      }
+      const localSelectedLayer = localState['selected'].value;
+      const localSelectedLayerId = Object.keys(localSelectedLayer)[0];
+
+      // Open form and get user input
+      const formValues = await new Promise<IDict>(resolve => {
+        const dialog = new ProcessingFormDialog({
+          title: 'Buffer',
+          schema: schema,
+          model: tracker.currentWidget?.model as IJupyterGISModel,
+          sourceData: {
+            InputLayer: localSelectedLayerId,
+            bufferDistance: 0,
+            projection: 'EPSG:4326',
+            attribution: ''
```

Comment:
I'd remove the `attribution` from the schema, this entry is there when using data sources from external services.

```suggestion
```